lib/ldaputil/certmap.c

```diff
@@ -757,23 +829,29 @@ ldapu_cert_searchfn_default(void *cert, LDAP *ld, void *certmap_info_in, const c
 
     if (certmap_info && certmap_info->searchAttr) {
         char *subjectDN = 0;
-        char *certFilter = 0;
-        int len;
 
         rv = ldapu_get_cert_subject_dn(cert, &subjectDN);
-
         if (rv != LDAPU_SUCCESS || !subjectDN) {
             return rv;
         }
-        len = strlen(certmap_info->searchAttr) + strlen(subjectDN) +
-              strlen("=") + 1;
-        certFilter = (char *)ldapu_malloc(len * sizeof(char));
+        size_t subjectDN_len = strlen(subjectDN);
+        size_t eqout_len = 3 * subjectDN_len + 1; /* big enough for worst case (and final \0) */
+        size_t searchAttr_len = strlen(certmap_info->searchAttr);
+        size_t len = searchAttr_len + 1 + eqout_len;
+        char *certFilter = (char *)ldapu_malloc(len * sizeof(char));
```

Comment:
It's not about your code, but still, we can fix it here;- shouldn't the check be `if (!(*certFilter)) {`?
As we need to check if the value at the address `certFilter` points to is NULL (i.e., if the allocation failed).

Reply:
???  why testing the address of a local variable ?

Reply:
Okay, I think I got a bit confused here... All good!